package/kernel/mac80211/patches/ath10k/990-ath10k-small-buffers.patch

```diff
@@ -1,64 +1,175 @@
---- a/drivers/net/wireless/ath/ath10k/htt.h
-+++ b/drivers/net/wireless/ath/ath10k/htt.h
-@@ -235,7 +235,11 @@ enum htt_rx_ring_flags {
+From ef2f5a3e1d761b4b73c926156bb284aef0d5817c Mon Sep 17 00:00:00 2001
+From: Nick Hainke <vincent@systemli.org>
+Date: Tue, 20 Dec 2022 17:39:39 +0100
+Subject: [PATCH] wifi: ath10k add smallbuffers module parameter
+
+The oom-killer gets triggered frequently by devices with small
+RAM (64 or 128 MB). This patch introduces a new module parameter
+"smallbuffers". Enabling the parameter reduces the buffer sizes
+used by the driver resulting in overall less memory usage.
+
+Signed-off-by: Nick Hainke <vincent@systemli.org>
+---
+ drivers/net/wireless/ath/ath10k/core.c   | 38 ++++++++++++++++++++++++
+ drivers/net/wireless/ath/ath10k/core.h   |  7 +++++
+ drivers/net/wireless/ath/ath10k/htt_rx.c |  5 +++-
+ drivers/net/wireless/ath/ath10k/pci.c    | 29 ++++++++++++++++++
+ 4 files changed, 78 insertions(+), 1 deletion(-)
+
+--- a/drivers/net/wireless/ath/ath10k/core.c
++++ b/drivers/net/wireless/ath/ath10k/core.c
+@@ -37,6 +37,9 @@ static bool uart_print;
+ static bool skip_otp;
+ static bool fw_diag_log;
+ 
++static unsigned int ath10k_smallbuffers_mode = ATH10K_SMALLBUFFERS_AUTO;
++bool ath10k_smallbuffers;
++
+ /* frame mode values are mapped as per enum ath10k_hw_txrx_mode */
+ unsigned int ath10k_frame_mode = ATH10K_HW_TXRX_NATIVE_WIFI;
+ 
+@@ -49,6 +52,7 @@ module_param_named(cryptmode, ath10k_cry
+ module_param(uart_print, bool, 0644);
+ module_param(skip_otp, bool, 0644);
+ module_param(fw_diag_log, bool, 0644);
++module_param_named(smallbuffers, ath10k_smallbuffers_mode, uint, 0644);
+ module_param_named(frame_mode, ath10k_frame_mode, uint, 0644);
+ module_param_named(coredump_mask, ath10k_coredump_mask, ulong, 0444);
+ 
+@@ -60,6 +64,7 @@ MODULE_PARM_DESC(frame_mode,
+ 		 "Datapath frame mode (0: raw, 1: native wifi (default), 2: ethernet)");
+ MODULE_PARM_DESC(coredump_mask, "Bitfield of what to include in firmware crash file");
+ MODULE_PARM_DESC(fw_diag_log, "Diag based fw log debugging");
++MODULE_PARM_DESC(smallbuffers, "Reduce buffersize (0: auto, 1: on, 2: off");
```

Comment:
I don't like 2 assigned to off...

I would rename the param to a something like 
buffers_mode 
and assign 1 to normal and 2 to reduced?

ath11k have a similar concept with wmi struct reduced to 2kb instead of 4kb so this pattern... so we can think of reusing this concept also there.